tests/e2e/specs/stories-editor/code-editor.test.js

```diff
@@ -0,0 +1,46 @@
+/**
+ * WordPress dependencies
+ */
+import { createNewPost, switchEditorModeTo } from '@wordpress/e2e-test-utils';
+
+/**
+ * Internal dependencies
+ */
+import { activateExperience, deactivateExperience } from '../../utils';
+
+describe( 'Code Editor', () => {
+	beforeAll( async () => {
+		await activateExperience( 'stories' );
+	} );
+
+	afterAll( async () => {
+		await deactivateExperience( 'stories' );
+	} );
+
+	it( 'allows switching to code editor and back', async () => {
+		await createNewPost( { postType: 'amp_story' } );
+
+		await page.waitForSelector( '#amp-story-block-navigation' );
+
+		await expect( page ).not.toMatchElement( '.block-editor-writing-flow__click-redirect' );
+
+		await switchEditorModeTo( 'Code' );
+		await page.click( '.edit-post-more-menu [aria-label="More tools & options"]' );
```

Comment:
Could also use `clickButtonByLabel` here which also waits until the selector becomes available, although its availability is not really an issue here.